src/pocket/PocketALBApplication.ts

```diff
@@ -143,35 +159,26 @@ export class PocketALBApplication extends Resource {
 
   /**
    * Creates the ALB stack and certificates
-   * @param config
-   * @param pocketVPC
    * @private
    */
-  private createALB(
```

Comment:
i'm guessing the reasoning behind minimizing params on these functions and leveraging a global `config` is less code, but i would argue this makes the functions more brittle and tougher to follow.

i know it's more code to explicitly define what data a particular function needs, but i think doing so makes the function easier to read and understand.

honestly, i think our design here is bad. these functions do too much. my vote here is to revert any config changes and make a ticket to refactor this file into smaller, more specialized functions.

happy to discuss further.

Reply:
I don't agree this makes the functions harder to follow, these are class methods, and I believe it's ok and for me even expected to use class properties within them, especially if they are properties that are accepted through the classes constructor. I agree we should discuss further, possibly in slack?

I'm ok with refactoring into more specialized functions, I can go either way there for this module.

Reply:
yeah, i think this is worth a discussion. i don't want to block progress on this repo, but i also don't want us to get too far along without assessing our patterns. i'll leave this up to the next reviewer and will approve if we have consensus. i'll also try to write up an issue where we can discuss async.